cli/exchange/node.go

```diff
@@ -14,13 +14,42 @@ type ExchangeNodes struct {
 	Nodes     map[string]interface{} `json:"nodes"`
 }
 
-func NodeList(org string, userPw string, node string, namesOnly bool) {
-	cliutils.SetWhetherUsingApiKey(userPw)
+// find correct credentials to use. Use -s or -n if one of them is not empty.
+// If both are empty, use HZN_EXCHANGE_USER_AUTH first, if it is not set use HZN_EXCHANGE_NODE_AUTH.
+func getCred(userPw string, nodeIdTok string) string {
+	credToUse := ""
+
+	if userPw != "" {
+		credToUse = userPw
+	} else {
+		if nodeIdTok != "" {
+			credToUse = nodeIdTok
+		} else {
+			if tmpU := cliutils.WithDefaultEnvVar(&userPw, "HZN_EXCHANGE_USER_AUTH"); *tmpU != "" {
```

Comment:
HZN_EXCHANGE_USER_AUTH is already read in hzn.go and passed in as userPw if set, so there is no need to check it here.

Reply:
In this case it is not read from hzn.go. The one that read in hzn.go are for other commands that will exit if no userPW or HZN_EXCHANGE_USER_AUTH are specified. Here we allow it to be empty.